src/Workspaces/Core/Portable/LanguageServices/FixAllSpanMappingService/IFixAllSpanMappingService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.FixAll
+{
+    /// <summary>
+    /// Language service for mapping spans for specific FixAllScopes for fix all occurences code fix.
+    /// Every language that wants to support span based FixAll scopes, such as FixAllScope.ContainingMember,
+    /// FixAllScope.ContainingType, should implement this language service. Non-span based FixAll scopes,
+    /// such as FixAllScope.Document, FixAllScope.Project and FixAllScope.Solution
+    /// do not require such a span mapping, and this service will never be called for these scopes. This language service
+    /// does not need to be implemented by languages that only intend to support these non-span based FixAll scopes.
+    /// </summary>
+    internal interface IFixAllSpanMappingService : ILanguageService
```

Comment:
Moved from shared layer to Workspaces layer as (1) this service is not needed in CodeStyle layer and (2) `CodeRefactorings.FixAllScope` is not accessible in CodeStyle layer